modules/azure/nsg.go

```diff
@@ -0,0 +1,194 @@
+package azure
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-09-01/network"
+)
+
+// GetDefaultNsgRulesClientE returns a rules client
+func GetDefaultNsgRulesClientE(subscriptionID string) (network.DefaultSecurityRulesClient, error) {
+	// Validate Azure subscription ID
+	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	if err != nil {
+		return network.DefaultSecurityRulesClient{}, err
+	}
+
+	nsgClient := network.NewDefaultSecurityRulesClient(subscriptionID)
+
+	// Get an authorizer
+	auth, err := NewAuthorizer()
+	if err != nil {
+		return network.DefaultSecurityRulesClient{}, err
+	}
+
+	nsgClient.Authorizer = *auth
+	return nsgClient, nil
+}
+
+// GetCustomNsgRulesClientE returns a rules client
```

Comment:
Please expand comment to explain what is "custom" about this client and how it differs from `GetDefaultNsgRulesClientE`.

Reply:
Added comments to explain difference between "default" and "custom"